Helper/Cart.php

```diff
@@ -1963,15 +1963,18 @@ public function collectDiscounts(
                     switch ($rule->getCouponType()) {
                         case RuleInterface::COUPON_TYPE_SPECIFIC_COUPON:
                         case RuleInterface::COUPON_TYPE_AUTO:
+                            $couponCode = $quote->getCouponCode();
+                            $description = trim(__('Discount ') . $rule->getDescription());
                             $discounts[] = [
-                                'description'       => trim(__('Discount ') . $rule->getDescription()),
+                                'description'       => $description,
                                 'amount'            => $roundedAmount,
-                                'reference'         => $quote->getCouponCode(),
+                                'reference'         => $couponCode,
                                 'discount_category' => Discount::BOLT_DISCOUNT_CATEGORY_COUPON,
-                                'discount_type'     => $this->discountHelper->convertToBoltDiscountType($quote->getCouponCode()), // For v1/discounts.code.apply and v2/cart.update
-                                'type'              => $this->discountHelper->convertToBoltDiscountType($quote->getCouponCode()), // For v1/merchant/order
+                                'discount_type'     => $this->discountHelper->convertToBoltDiscountType($couponCode), // For v1/discounts.code.apply and v2/cart.update
+                                'type'              => $this->discountHelper->convertToBoltDiscountType($couponCode), // For v1/merchant/order
                             ];            
-
+                            $this->logEmptyDiscountCode($couponCode, $description);
```

Comment:
Looks like we shouldn't bugsnag for RuleInterface::COUPON_TYPE_AUTO, because we already know that couponCode is empty for them and it's ok.

Reply:
@vitaliyreznikov For RuleInterface::COUPON_TYPE_AUTO, it still has code by default.

Reply:
Could you please provide example? How it looks for example for default rule "buy 3 get 4"?

Reply:
@vitaliyreznikov  "buy 3 get 4" belong the rule type COUPON_TYPE_NO_COUPON not COUPON_TYPE_AUTO

Reply:
Got it. Anyway, can we skip bugsnag for this case?

Reply:
What do you mean? for which case? RuleInterface::COUPON_TYPE_AUTO or COUPON_TYPE_NO_COUPON? Currently in this PR we already ignore bugsnag for COUPON_TYPE_NO_COUPON.